components/homme/cmake/HommeMacros.cmake

```diff
@@ -123,7 +123,12 @@ macro(createTestExec execName execType macroNP macroNC
   # Add this executable to a list
   SET(EXEC_LIST ${EXEC_LIST} ${execName} CACHE INTERNAL "List of configured executables")
 
-  TARGET_LINK_LIBRARIES(${execName} pio timing ${COMPOSE_LIBRARY} ${BLAS_LIBRARIES} ${LAPACK_LIBRARIES})
+  if(BUILD_HOMME_WITHOUT_PIOLIBRARY)
+    TARGET_LINK_LIBRARIES(${execName} timing ${COMPOSE_LIBRARY} ${BLAS_LIBRARIES} ${LAPACK_LIBRARIES})
+  ELSE()
+    TARGET_LINK_LIBRARIES(${execName} pio timing ${COMPOSE_LIBRARY} ${BLAS_LIBRARIES} ${LAPACK_LIBRARIES})
```

Comment:
So that there aren't two lines with lists have N-1 redundant entries, you could make a list with the N-1 entries, then use `list(APPEND ...)` for pio.